packages/firestore/src/local/remote_document_change_buffer.ts

```diff
@@ -67,8 +68,14 @@ export abstract class RemoteDocumentChangeBuffer {
 
   /** Buffers a `RemoteDocumentCache.addEntry()` call. */
   addEntry(maybeDocument: MaybeDocument): void {
-    const changes = this.assertChanges();
-    this.changes = changes.insert(maybeDocument.key, maybeDocument);
+    this.assertChanges();
+    this.changes!.set(maybeDocument.key, maybeDocument);
```

Comment:
I can go back to this. The main reason I removed this was that we now directly modify the instance variable and I thought it was cleaner to make this explicit.

Reply:
FWIW- It might feel slightly cleaner to have a nullable `_changes` property and then make `changes` a non-nullable getter that does the assert, similar to https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/memory_persistence.ts#L215